File: Dataset/cs/Calculate.cs

//  Calculate.cs  (c) 2003 Kari Laitinen

using System ;

class Calculate
{
   static void print_calculations ( int first_integer,
                                    int second_integer )
   {
      Console.Write( "\n    "  +  first_integer  +  " + "
                  +  second_integer  +  " = "
                  +  ( first_integer  +  second_integer ) 
                  +  "\n    "  +  first_integer  +  " - "
                  +  second_integer  +  " = "
                  +  ( first_integer - second_integer )
                  +  "\n    "  +  first_integer  +  " * "
                  +  second_integer  +  " = "
                  +  first_integer * second_integer ) ;

      if ( second_integer  !=  0 )
      {
         Console.Write( "\n    "  +  first_integer  +  " / "
                     +  second_integer  +  " = "
                     +  first_integer / second_integer
                     +  "\n    "  +  first_integer  +  " % "
                     +  second_integer  +  " = "
                     +  first_integer % second_integer  +  "\n" ) ;
      }
      else
      {
         Console.Write( "\n    Cannot divide with zero. \n" ) ;
      }
   }

   static void Main( string[] command_line_parameters )
   {
      int  first_operand, second_operand ;

      if ( command_line_parameters.Length  ==  2 )
      {
         first_operand   =  Convert.ToInt32( command_line_parameters[ 0 ] ) ;
         second_operand  =  Convert.ToInt32( command_line_parameters[ 1 ] ) ;

         print_calculations( first_operand, second_operand ) ;
      }
      else
      {
         Console.Write( "\n This program calculates with integers."
                     +  "\n\n   Give first integer:   " ) ;
         first_operand   =  Convert.ToInt32( Console.ReadLine() ) ;
         Console.Write( "   Give second integer:  " ) ;
         second_operand  =  Convert.ToInt32( Console.ReadLine() ) ;

         print_calculations( first_operand, second_operand ) ;
      }
   }
}

File: Dataset/cpp/calculate.cpp

//  calculate.cpp (c) 1998-2002 Kari Laitinen

#include  <iostream.h>
#include  <stdlib.h>

void  print_calculations ( int first_integer,
                           int second_integer )
{
   cout << "\n    " << first_integer << " + "
        << second_integer << " = "
        << first_integer + second_integer 
        << "\n    " << first_integer << " - "
        << second_integer << " = "
        << first_integer - second_integer
        << "\n    " << first_integer << " * "
        << second_integer << " = "
        << first_integer * second_integer ;

   if ( second_integer  !=  0 )
   {
      cout << "\n    " << first_integer << " / "
           << second_integer << " = "
           << first_integer / second_integer
           << "\n    " << first_integer << " % "
           << second_integer << " = "
           << first_integer % second_integer << "\n" ;
   }
   else
   {
      cout << "\n    Cannot divide with zero. \n" ;
   }
}

int main( int    number_of_command_line_arguments,
          char*  command_line_arguments[] )
{
   int  first_operand, second_operand ;

   if ( number_of_command_line_arguments  ==  3 )
   {
      first_operand   =  atoi( command_line_arguments[ 1 ] ) ;
      second_operand  =  atoi( command_line_arguments[ 2 ] ) ;

      print_calculations( first_operand, second_operand ) ;
   }
   else
   {
      cout << "\n This program can perform basic calculations."
           << "\n Give two integers separated by a space:  " ;

      cin  >> first_operand  >>  second_operand ;

      print_calculations( first_operand, second_operand ) ;
   }
}



